docs/content/docs/dev/datastream/fault-tolerance/serialization/types_serialization.md

```diff
@@ -359,12 +385,13 @@ You can still use the same method as in Java as a fallback.
 There are two ways to create a TypeSerializer.
 
 The first is to simply call `typeInfo.createSerializer(config)` on the `TypeInformation` object.
-The `config` parameter is of type `ExecutionConfig` and holds the information about the program's registered
-custom serializers. Where ever possibly, try to pass the programs proper ExecutionConfig. You can usually
-obtain it from `DataStream` via calling `getExecutionConfig()`.
+The `config` parameter is of type `SerializerConfig` and holds the information about the program's registered
```

Comment:
Need double-check: why do we need to create type serializer in user code? If not, we can delete the relevant content here

Reply:
Now that it is supported to register a custom `TypeInfoFactory`, I think we should discourage the use case here and remove the relevant section in doc. WDYT?